src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -25,6 +25,10 @@ open Statement_gen
 
 let standalone_functions = ref false
 
+type found_functor = {template: string; signature: string; defn: string}
+
+let (functors : (string, found_functor list) Hashtbl.t) = String.Table.create ()
```

Comment:
I'm not totally following why this has to be module level object. Can you just make it in `pp_prog` and pass it as an argument? Would we at some point want to use `Stan_math_code_gen.functors` outside of `pp_prog`?

Reply:
Fair point, I'll move it inside